pdns/recursordist/lwres.cc

```diff
@@ -423,6 +576,9 @@ static LWResult::Result asyncresolve(const ComboAddress& address, const DNSName&
   bool weWantEDNSSubnet = false;
   uint8_t outgoingECSBits = 0;
   ComboAddress outgoingECSAddr;
+  std::optional<ComboAddress> addressToBindTo;
+  std::optional<EDNSCookiesOpt> cookieSentOut;
+
   if (EDNS0Level > 0) {
```

Comment:
do I understand correctly that we don't even try cookies until we have established EDNS support?

Reply:
I'll do a double check of this logic soon.

Reply:
Since some time we start with in with EDNS0Level 1, so we initially assume EDNS is supported (see `LWResult::Result SyncRes::asyncresolveWrapper`).